pkg/controller/jobframework/reconciler.go

```diff
@@ -428,8 +428,15 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 				log.V(3).Info("Handling job with no workload found an existing workload")
 				return ctrl.Result{Requeue: true}, nil
 			}
+			// else if errors.Is(err, ErrPrebuildWorkloadNotFound) {
+			// TODO: ???
+			// log.V(3).Info("Handling job with no prebuild workload. Waiting one second...")
+			// return ctrl.Result{RequeueAfter: time.Second}, nil
+			// }
 			if IsUnretryableError(err) {
 				log.V(3).Info("Handling job with no workload", "unretryableError", err)
+				// TODO: If the error is non-retryable, should we return without an error?
```

Comment:
Same here, I don't understand why / if we need these TODO changes.

Reply:
You're right. This is not related to this PR. I'll revert it.

My only question is, shouldn't we ignore unretryable errors?

Reply:
I'm not sure - my question is - what is the user-facing or admin-facing difference? In other words: what is the issue the code aims to solve. Otherwise, it is really hard for me to answer.